pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOrderByOperator.java

```diff
@@ -143,24 +167,134 @@ public SelectionOrderByOperator(IndexSegment indexSegment, QueryContext queryCon
 
   @Override
   protected IntermediateResultsBlock getNextBlock() {
-    TransformBlock transformBlock;
-    while ((transformBlock = _transformOperator.nextBlock()) != null) {
-      int numExpressions = _expressions.size();
+    int numExpressions = _expressions.size();
+    int numOrderByExpressions = _orderByExpressions.size();
+
+    if (numExpressions == numOrderByExpressions) {
```

Comment:
not sure if this a valid check. what if the expressions are functions?

Reply:
Both order-by and selected expressions can be identifiers or functions. Expressions here are order-by expressions followed by non-order-by expressions (deduplicated, see `SelectionOperatorUtils.extractExpressions()` for more details.